sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,538 @@
+while there are any `async` invocations for which `get` has not been called.
+
+== `task_sequence` Properties
+
+The following code and table describe the properties that can be provided when
+declaring a `task_sequence` object.
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+struct balanced_key {
+  using value_t = property_value<balanced_key>;
+};
+
+struct invocation_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<invocation_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+struct response_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<response_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+inline constexpr balanced_key::value_t balanced;
+template <uint32_t Size>
+inline constexpr invocation_capacity_key::value_t<Size> invocation_capacity;
+template <uint32_t Size>
+inline constexpr response_capacity_key::value_t<Size> response_capacity;
+
+template <> struct is_property_key<balanced_key> : std::true_type {};
+template <> struct is_property_key<invocation_capacity_key> : std::true_type {};
+template <> struct is_property_key<response_capacity_key> : std::true_type {};
+
+template <auto &f, typename PropertyListT>
+struct is_property_key_of<balanced_key, task_sequence<f, PropertyListT>>
+  : std::true_type {};
+template <auto &f, typename PropertyListT>
+struct is_property_key_of<invocation_capacity_key, task_sequence<f, PropertyListT>>
+  : std::true_type {};
+template <auto &f, typename PropertyListT>
+struct is_property_key_of<response_capacity_key, task_sequence<f, propertiesT>>
+  : std::true_type {};
+
+} // namespace sycl::ext::intel::experimental
+----
+
+--
+[options="header"]
+|===
+| Property | Description
+| balanced | The `balanced` property is a guarantee to the SYCL device compiler
+that the `task_sequence` object will call exactly the same number of `async` s
+and `get` s over the object's lifetime (i.e. before the `task_sequence`
+destructor is invoked).
+
+| invocation_capacity | The `async` invocations are guaranteed to not block the
+caller as long as the number of outstanding invocations are less than or equal
+to `invocation_capacity`. An outstanding invocation is an inflight
+`task_sequence` invocation that has not yet completed.
+
+A default value is chosen by the implementation.
+
+| response_capacity | A `task_sequence` invocation is allowed to write its
+results and completion status to the results data structure of the
+`task_sequence` if there is sufficient capacity to accommodate it. A
+`response_capacity` of N indicates that the results
+data structure should be sized such that the oldest N invocations of the `task_sequence` can be
+successfully written to the results data structure. When an invocation is able
+to write its result to the results data structure, it can transition from an
+outstanding state to a completed state. At least one outstanding `async` call
```

Comment:
It seems the problematic definition of outstanding that I called out above is pervasive.  Perhaps we just shouldn't use the term "outstanding" in the definition of invocation capacity.